demonstrations/tutorial_zne_catalyst.py

```diff
@@ -0,0 +1,282 @@
+computations. First introduced in [#temme2017zne]_, it helps improve the accuracy of quantum
+results by running circuits at varying noise levels and extrapolating back to a hypothetical
+zero-noise case. While this tutorial won't delve into the theory behind ZNE in detail, let's first
+review what happens when using the protocol in practice.
+
+Stage 1: Generating Noise-Scaled Circuits
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In its digital version [#DZNEpaper]_, ZNE works by generating circuits with **increased** noise. 
+Currently, ZNE in Catalyst supports two methods for generating noise-scaled circuits:
+
+1. **Global folding**: If a circuit implements a global unitary :math:`U`, global folding applies
+   :math:`U(U^\dagger U)^n` for some integer :math:`n`,
+   effectively scaling the noise in the entire circuit.
+2. **Local folding**: Individual gates are repeated (or folded) in contrast with the entire
+   circuit.
+
+Stage 2: Running the circuits
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once noise-scaled circuits are created, they need to be run! These can be executed on either real
+quantum hardware or a noisy quantum simulator. In this tutorial, we'll use the
+`Qrack quantum simulator <https://qrack.readthedocs.io/>`_, which is both compatible with Catalyst,
+and implements a noise model. For more about the integration of Qrack and Catalyst, see
+the demo `QJIT compilation with Qrack and Catalyst <https://pennylane.ai/qml/demos/qrack/>`_.
+
+Stage 3: Combining the results
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+After executing the noise-scaled circuits, an extrapolation on the results is performed  
+to estimate the zero-noise limit---the result we would expect in a noise-free scenario. 
+Catalyst provides two methods for perfoming this extrapolation:
+
+1. **Polynomial extrapolation**, and
+2. **Exponential extrapolation**.
+
+Using ZNE with Pennylane
+------------------------
+The demo `Error mitigation with Mitiq and PennyLane <https://pennylane.ai/qml/demos/tutorial_error_mitigation/>`_
+shows how ZNE, along with other error mitigation techniques, can be carried out in Pennylane by using Mitiq, 
+a Python library developed by Unitary Fund.
+
+ZNE in particular is also offered out of the box in Pennylane as a *differentiable* error mitigation technique,
+for usage in combination with variational workflows. More on this in the tutorial 
+`Differentiating quantum error mitigation transforms <https://pennylane.ai/qml/demos/tutorial_diffable-mitigation/>`_.
+
+
+Using ZNE in Catalyst
+---------------------
+On top of the error mitigation routines offered in Pennylane, ZNE is also available for just-in-time 
+(JIT) compilation, starting from Catalyst v0.8.1.
+In this tutorial we see how an error mitigation routine can be integrated in a Catalyst workflow.
+
+At the end of the tutorial, we will compare time for the execution of ZNE routines in 
+pure Pennylane vs. Pennylane Catalyst with JIT. 
+
+Mirror circuit
+--------------
```

Comment:
```suggestion
Defining the mirror circuit
---------------------------
```

Maybe a bit more explanatory?